biorbd_optim/dynamics.py

```diff
@@ -63,6 +63,32 @@ def forward_dynamics_torque_muscle_driven(states, controls, nlp):
         qddot_reduced = nlp["q_dot_mapping"].reduce.map(qddot)
         return vertcat(qdot_reduced, qddot_reduced)
 
+    @staticmethod
+    def forward_dynamics_excitation_driven(states, controls, nlp):
+        q, qdot, qdot_reduced, residual_tau = Dynamics.__dispatch_data(states, controls, nlp)
+
+        muscles_states = biorbd.VecBiorbdMuscleStateDynamics(nlp["nbMuscle"])
+        muscles_excitation = controls[nlp["nbTau"]:]
+        muscles_activations = states[nlp["nbQ"] + nlp["nbQdot"]:]
+
+        for k in range(nlp["nbMuscle"]):
+            muscles_states[k].Excitation(muscles_excitation[k])
+            muscles_activations[k] = muscles_states[k].activation()
+
+        # muscles_activations =
+
+        for k in range(nlp["nbMuscle"]):
+            muscles_states[k].setActivation(muscles_activations[k])
+        muscles_tau = nlp["model"].muscularJointTorque(muscles_states, q, qdot).to_mx()
+
+        tau = muscles_tau + residual_tau
+
+        qddot = biorbd.Model.ForwardDynamics(nlp["model"], q, qdot, tau).to_mx()
+        qddot_reduced = nlp["dof_mapping"].reduce(qddot)
```

Comment:
This is no longer true. It should read:
`qddot_reduced = nlp["qdot_mapping"].reduce.map(qddot)`